new/resource_lifecycle.md

```diff
@@ -0,0 +1,119 @@
+3. Should be well-designed and self contained.
+4. Should be tested.
+5. Should be maintained on [supported platforms](https://github.com/chef/chef-rfc/blob/master/rfc021-platform-support-policy.md) as relevant.
+4. Should minimize surprises.
+
+At any given time current resources have an implicit state. To clarify and be explicit about these states, it is proposed that resources have the following states:
+
+#### Analysis
+
+Resources in analysis state are being evaluated to identify who, what, when, and why a resource is (still) needed. A resource could be in this state whether it's a popular widely used resource in a community cookbook or a gap in common infrastructure element. A resource could re-enter this state when it has been identified as a problematic resource that needs to be evaluated for continued support.
+
+#### Development
+
+Resources in development state are in active development. The value of adding the resource is understood. The number of impacted users has been estimated.
+
+#### Supported
+
+Resources in supported state have been added to Chef, work on relevant platforms, and are tested.
+
+#### Deprecated
+
+Resources in deprecated state have been identified as having significant issues that can not or will not be resolved. Use of the resource will emit a deprecation warning. A compatible cookbook(s) for the deprecated resource may be created or updated to allow use of the deprecated resource.
+
+#### Retired
+
+Resources in retired state have been in deprecated state and emitting warnings and on the next major release of Chef removed from code.
+
+## Specification
+### Criteria for adopting resources
+
+Factors that influence and inform the decision to adopt a resource include:
+
+* widespread use of a resource based on analysis of publicly available supermarket cookbooks,
+* well tested and supported cookbook resources,
+* percentage of supported platforms,
+* impact of implementing as a primative that can be built upon.
+
+### Criteria for deprecating resources
+
+Factors that influence and inform the decision to deprecate a resource include:
+
+* limited use based on analysis of publicly available supermarket cookbooks,
+* outdated code or lack of support based on quality of resource and its usability,
+* security implications,
+* complex implementation that lacks clarity of consequences of use.
+* impact of using the resource incorrectly.
+
+### Process for adopting resources
+
+* Identify resources for adoption.
+* Create an RFC following the [Chef RFC](https://github.com/chef/chef-rfc) process.
+* Announce on Chef mailing list and Chef Community Slack.
+* Any proposed changes to the interfaces must be implemented in the cookbook prior to adoption.
+* Any bugs discovered must be repaired in the cookbook and released prior to adoption.
+* Resource is added to core chef and documentation is updated.
+* Add deprecated cookbook warnings for conflicting cookbooks.
```

Comment:
If the API of the resource needs to change in core compared to the cookbook then the resource should be renamed in core so as not to conflict with the cookbook.

an example is the `chocolatey_package` resource, which needed to be renamed anyway to comply with the naming standard that we have in core, and it needed to adopt the API shape of the rest of the package resources.  the cookbook version of the resource was designed around chocolatey's CLI API, while the core version of the resource was designed around core chef's package resource API, which was fundamentally different design concerns.  since they were named differently, they did not conflict.

Reply:
At what point would we change the resource name? You mean upon adoption or post adoption?

Reply:
If the naming structure doesn't match in the cookbook we'd change it as we bring it into core. That wouldn't impact the cookbook.